reconcile/utils/oc.py

```diff
@@ -1194,40 +1252,31 @@ def get_all(self, kind, all_namespaces=False):
             raise StatusCodeError(f"[{self.server}]: {e}")
 
     @staticmethod
-    def add_group_kind(kind, kgv, new, preferred):
+    def add_api_resource(kind, kgv, preferred, resource):
```

Comment:
Parameters:

- `kgv` Please use `kind_groupversion` instead
- `preferred` Preferred what?

Reply:
`preferred` is part of the kubernetes client we are using, so I think it's ok to use the same.

I agree with `kgv`one. As this is now the `api_resources` dictionary, maybe it is better to use `api_resources` directly. WDYT? 

I'm also wondering if the `copy` is really needed. I tried to reuse the existing code as much as I could, but this might not be necessary.

Reply:
I refactored this a bit using the same names as the Kubernetes dynamic client. PTAL